docs/docs/product/users.md

```diff
@@ -0,0 +1,129 @@
+
+## Permissions
+
+An admin can grant or remove permissions on any database or schema to any user with the role of `User`. Mathesar currently supports the following permissions:
+
+Database Permissions
+
+1. Manager
+2. Editor
+3. Viewer
+
+Schema Permissions
+
+1. Manager
+2. Editor
+3. Viewer
+
+### Actions associated with each permission
+
+**Database Permissions**
+
+| Permission                                  | Database Manager        | Database Editor | Database Viewer |
+| ------------------------------------------- | ----------------------- | --------------- | --------------- |
+| Grant or Revoke permissions on the database | &#9745;                 | -               | -               |
+| Add and remove schemas in the database      | &#9745;                 | -               | -               |
+| Permissions on all contained schemas        | Manager, Editor, Viewer | Editor, Viewer  | Viewer          |
+
+**Schema Permissions**
+
+| Permission                                | Schema Manager | Schema Editor | Schema Viewer |
+| ----------------------------------------- | -------------- | ------------- | ------------- |
+| Grant or revoke permissions on the schema | &#9745;        | -             | -             |
+| Add and remove tables in the schema       | &#9745;        | -             | -             |
+| Add and remove shared explorations        | &#9745;        | &#9745;       | -             |
+
+### Order or precedence of permissions
+
+TODO:
+
+## Add a new user
+
+> Requires `Admin` role.
+
+1. Click on the gear icon on the top right of the application in the navigation bar to open a dropdown menu.
+2. Click on the `Administration` option.
+3. In the left sidebar click on the `Users` option.
+4. Click on the `Add User` button, right next to the search bar.
+5. Fill in the form with the required details.
+6. You can use any password for now. The user will be prompted to change the password when he/she logs in for the first time.
+7. Choose the role as per the explanation here: [Roles](#roles)
+8. Click on the `Save` button.
+
+## Editing a user's details and role
+
+> Requires `Admin` role.
+
+1. Click on the gear icon on the top right of the application in the navigation bar to open a dropdown menu.
+2. Click on the `Administration` option.
+3. In the left sidebar click on the `Users` option.
+4. From the list click on the user whose details you want to edit.
+5. Edit the desired details. You can even change the role.
+6. Click on the `Save` button.
+
+## Deleting a user
+
+> Requires `Admin` role.
+
+1. Click on the gear icon on the top right of the application in the navigation bar to open a dropdown menu.
+2. Click on the `Administration` option.
+3. In the left sidebar click on the `Users` option.
+4. From the list click on the user who you want to delete.
+5. Click on the `Delete User` button.
+6. In the confirmation modal, again click on the `Delete User` button.
+
+## Granting database-level permissions
```

Comment:
@kgodey do we want to add steps for editing permissions too? As of now UI doesn't support that feature, the user will have to delete the current permissions and grant the new permission.

Reply:
No, it's fine, we can mention that the user will need to delete current permissions and add a new permission.